### ISSUE TITLE
Adding additional directories generates UnCaught TypeError

### ISSUE THREAD

**Issue by EdmundsEcho** [coryhouse/ps-react#13]
When I create a new component directory (with or without the accompanying docs directory), I generate an error after restarting the server.
I expect that after restarting the server, to see the new component listed in the docs page.
Instead I receive the following error.
![screen shot 2017-08-17 at 4 21 06 pm](https://user-images.githubusercontent.com/11245417/29432069-27c6be4c-8368-11e7-9c0d-b967832e5cf9.png)

I can restore the page by removing the new directory and restarting the server.  I'm using the configuration files "as is" from this repo.

**Comment by coryhouse** [comment]
Correct, that's as designed. If you create a component folder without placing a component with the same name inside, then it will throw an error. I say it's by design because that's the way the project is intended to be used (though in retrospect, I could certainly have provided this helpful explanation when this issue occurs...)

**Comment by EdmundsEcho** [comment]
Thank you for the prompt reply. I included a file with the same name... 

![screen shot 2017-08-17 at 4 54 34 pm](https://user-images.githubusercontent.com/11245417/29433430-12deca60-836d-11e7-83ec-365967bace38.png)

and here are the files with the new/matching class names that successfully compile.

index.js
`export {default} from './ProgressBarE.js';

ProgressBarE.js
`import React from 'react';
import PropTypes from 'prop-types';

class ProgressBarE extends React.Component {

  // a component with a render statement that depends on
  // two functions...

  getColor = (percent) => {
    switch ( true ) {
      case ( this.props.percent === 100 ) : return 'green';
      case ( this.props.percent > 50 )    : return 'lightgreen';
      default                             : return 'red'
    }
  }

  getWidthAsPercentOfTotalWidth = () => {
    return parseInt(this.props.width * (this.props.percent / 100), 10);
  }

  render() {
    const {percent, width, height} = this.props;
    return (
      <div style={{border: 'solid 1px lightgray', width: width}}>
        <div style={{
          width: this.getWidthAsPercentOfTotalWidth(),
          height,
          backgroundColor: this.getColor(percent)
        }} />
      </div>
    );
  }
}

ProgressBarE.propTypes = {
  col: PropTypes.string.isRequired
}

export default ProgressBarE`

**Comment by coryhouse** [comment]
Can you share a fork of this repo that exhibits the issue? I can't reproduce this.

**Comment by EdmundsEcho** [comment]
Here you go: git@github.com:EdmundsEcho/ps-react.git

**Comment by coryhouse** [comment]
Interesting, thanks for sharing the solution. Out of curiosity, what were the hidden files? Just wondering if what happened to you could be a common issue...

**Comment by EdmundsEcho** [comment]
`.tern-port`  - This is part of my auto-completion set-up vim/neovim.

**Comment by ultrox** [comment]
@EdmundsEcho @coryhouse The same thing happened to me yesterday, it took me some time debugging, and even tho I rejected idea of file .tern file causing the problem I was successfully recreated situation. Go in both folders component and examples and create the invisible file, or any file for that matter, and see it fail.

It was very hard to find, as I wasn't sure what caused it because .tern wasn't creating its file when it wasn't triggered, and it needed to be in both directories. 

Anyway, after long hours debugging it's working now 👯, and as usual it one line of code.

**Comment by coryhouse** [comment]
To clarify, this issue is caused by placing any non-js files in `/src/components` or `/src/docs/examples`. PR #21 resolves this issue. I'm not going to merge the PR until I patch the course, so leaving this note here to help others in the meantime.
